docs/clients/fluentd/README.md

```diff
@@ -0,0 +1,209 @@
+```
+
+## Usage
+In your Fluentd configuration, use `@type loki`. Additional configuration is optional, default values would look like this:
+```
+<match **>
+  @type loki
+  url "https://logs-us-west1.grafana.net"
+  username "#{ENV['LOKI_USERNAME']}"
+  password "#{ENV['LOKI_PASSWORD']}"
+  extra_labels {"env":"dev"}
+  flush_interval 10s
+  flush_at_shutdown true
+  buffer_chunk_limit 1m
+</match>
+```
+
+### Using labels
+
+Simple label from top level attribute
+```
+<match mytag>
+  @type loki
+  # ...
+  <label>
+    fluentd_worker
+  </label>
+  # ...
+</match>
+```
+
+You can rewrite the label keys as well as the following
+
+```
+<match mytag>
+  @type loki
+  # ...
+  <label>
+    worker fluentd_worker
+  </label>
+  # ...
+</match>
+```
+
+You can use record accessor syntax for nested field. https://docs.fluentd.org/plugin-helper-overview/api-plugin-helper-record_accessor#syntax
+
+```
+<match mytag>
+  @type loki
+  # ...
+  <label>
+    container $.kubernetes.container
+  </label>
+  # ...
+</match>
+```
+
+### Extracting Kubernetes labels
+
+As Kubernetes labels are a list of nested key-value pairs there is a separate option to extract them.
+Note that special characters like "`. - /`" will be overwritten with `_`.
+Use with the `remove_keys kubernetes` option to eliminate metadata from the log.
+```
+<match mytag>
+  @type loki
+  # ...
+  extract_kubernetes_labels true
+  remove_keys kubernetes
+  <label>
+    container $.kubernetes.container
+  </label>
+  # ...
+</match>
+```
+
+### Multi-worker usage
+
+Loki doesn't currently support out-of-order inserts - if you try to insert a log entry an earlier timestamp after a log entry with with identical labels but a later timestamp, the insert will fail with `HTTP status code: 500, message: rpc error: code = Unknown desc = Entry out of order`. Therefore, in order to use this plugin in a multi worker Fluentd setup, you'll need to include the worker ID in the labels.
+
+For example, using [fluent-plugin-record-modifier](https://github.com/repeatedly/fluent-plugin-record-modifier):
+```
+<filter mytag>
+    @type record_modifier
+    <record>
+        fluentd_worker "#{worker_id}"
+    </record>
+</filter>
+
+<match mytag>
+  @type loki
+  # ...
+  <label>
+    fluentd_worker
+  </label>
+  # ...
+</match>
+```
+
+### Using multiple buffer flush threads
+
+Similarly, when using `flush_thread_count` > 1 in the [`buffer`](https://docs.fluentd.org/configuration/buffer-section#flushing-parameters)
+section, a thread identifier must be added as a label to ensure that log chunks flushed in parallel to loki by fluentd always have increasing
+times for their unique label sets.
+
+This plugin automatically adds a `fluentd_thread` label with the name of the buffer flush thread when `flush_thread_count` > 1.
+
+## Docker Image
+
+There is a Docker image `grafana/fluent-plugin-grafana-loki:master` which contains default configuration files to git log information
```

Comment:
Having difficulty understanding this line. Can you elaborate?

Reply:
I'm asking about this line because it's not clear for me https://grafana.slack.com/archives/CEPJRLQNL/p1580902179287300

let me try to elaborate with my understanding anyway

Reply:
♻️ https://github.com/grafana/loki/pull/1634/commits/704c303e1ea10d44abb72dd513adca8f4fc14d23